mantidproject/mantid
# Determine SANS Instrument from User File

**Description of work.**
Removed the instrument combo box from the main page of the SANS gui. Added a instrument parser which extracts instrument from the user file, and updates settings accordingly. Process button is disabled while instrument is set to (as is now the default) "NoInstrument".

**Report to:** [nobody]

**To test:**
On MantidPlot, open Interfaces->SANS->ISIS SANS v2 experimental
Looking to the instrument at the top of the main page, it should initially be the uneditable text "NoInstrument" and the Process button should be disabled.

Load in any acceptable User file (one which contains a line containing nothing but one of the SANS instruments: "SANS2D", "ZOOM", "LOQ", "LARMOR"). The instrument text should change to reflect this, and the process button should be enabled. Furthermore, the settings in settings tab should change to reflect the new instrument.

If you load a user file, which does not contain a recognised instrument, a warning box will be displayed.

Additionally, the following tests should pass:
`user_file_parser_test`
`user_file_reader_test`
`run_tab_presenter_test`

Fixes #23971 
Fixes #24053 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
